src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -829,6 +885,32 @@ protected function callScope($scope, $parameters)
     }
 
     /**
+     * Get the underlying query builder instance with applied global scopes.
+     *
+     * @return \Illuminate\Database\Query\Builder|static
+     */
+    public function getQueryWithScopes()
+    {
+        if (! $this->scopes) {
+            return $this->getQuery();
+        }
+
+        $builder = clone $this;
+
+        foreach ($this->scopes as $scope) {
+            if ($scope instanceof Closure) {
+                $scope($builder);
+            }
+
+            if ($scope instanceof ScopeInterface) {
```

Comment:
This if-if construct could be changed into if-elseif, however I still don't really like that as scopes that do not adhere to the interface would simply be skipped unknowingly. Perhaps introduce `callScope(ScopeInterface $scope)` and call it from an else-statement, so that PHP does the type checking for us.